eng/common/scripts/Verify-Links.ps1

```diff
@@ -257,21 +272,29 @@ while ($pageUrisToCheck.Count -ne 0)
 
   $linkUris = GetLinks $pageUri
   Write-Host "Found $($linkUris.Count) links on page $pageUri";
-  
+  $badLinksPerPage = @{};
   foreach ($linkUri in $linkUris) {
     $isLinkValid = CheckLink $linkUri
-    if (!$isLinkValid) {
-      $script:badLinks += $linkUri
+    if (!$isLinkValid -and !$badLinksPerPage.ContainsKey($linkUri)) {
+      $badLinksPerPage[$linkUri] = $true
     }
     if ($recursive -and $isLinkValid) {
       if ($linkUri.ToString().StartsWith($baseUrl) -and !$checkedPages.ContainsKey($linkUri)) {
         $pageUrisToCheck.Enqueue($linkUri);
       }
     }
   }
+  if ($badLinksPerPage.Count -gt 0) {
+    $badLinks[$pageUri] = $badLinksPerPage
+  }
 }
 
-Write-Host "Found $($checkedLinks.Count) links with $($badLinks.Count) broken"
-$badLinks | ForEach-Object { Write-Host "  $_" }
+Write-Host "Found $($checkedLinks.Count) links with $($badLinks.Count) pages broken."
+foreach ($pageLink in $badLinks.Keys) {
+  LogError "  '$pageLink' has $($badLinks[$pageLink].Count) broken link(s)`:"
```

Comment:
Please test this but I believe Devops might stop the process on the first error logged if that is the case we should either format this into one string and log it at once or we should show an error as the last line we output.

Reply:
Updated to description. Could you check which one is better?
Seems like the dev op log message is not that perfect for multiple lines.

Reply:
Do you have a log for a test run I can look here to see what it looks like?